homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,489 @@
+        return False
+
+    phone_id = config[DOMAIN].get(CONF_PHONE_ID)
+    device_id = config[DOMAIN].get(CONF_DEVICE_ID)
+    device_password = config[DOMAIN].get(CONF_DEVICE_PASSWORD)
+    include_sensors = config[DOMAIN].get(CONF_INCLUDE_SCHEDULE_SENSORS)
+    schedules_scan_interval = config[DOMAIN].get(CONF_SCHEDULE_SCAN_INTERVAL)
+
+    if None in [phone_id, device_id, device_password]:
+        _LOGGER.error("No details for device connection supplied")
+        return False
+
+    v2bridge = None
+
+    async def async_update_schedules_data(timestamp: Optional[datetime]) \
+            -> None:
+        """Update the schedules data."""
+        if SWITCH_ENTITY:
+            response = await get_schedules(
+                SWITCH_ENTITY.device_ip_addr, phone_id,
+                device_id, device_password)
+
+            if response.found_schedules and SENSOR_ENTITIES:
+                updated_entities = []  # type: List[SwitcherV2Schedule]
+
+                for schedule in response.get_schedules:
+                    current_entity = SENSOR_ENTITIES[schedule.schedule_id]
+
+                    await current_entity.async_update_schedule(schedule)
+
+                    updated_entities.append(current_entity)
+
+                for entity in SENSOR_ENTITIES.values():
+                    if entity not in updated_entities:
+                        await entity.async_deconfigure_schedule()
+
+                updated_entities = []
+
+        return None
+
+    async def async_initial_data_cb(device_data: SwitcherV2Device) -> None:
+        """Use this as callback for creating home assistant's entities.
+
+        Called by the bridge thread on initial device discovery.
+        """
+        discovery_info = {
+            DISCOVERY_CONFIG: config[DOMAIN],
+            DISCOVERY_DEVICE: device_data
+        }
+
+        # CONF_SCHEDULE_SCAN_INTERVAL is a timedelta and not serializable.
+        discovery_info[DISCOVERY_CONFIG].pop(CONF_SCHEDULE_SCAN_INTERVAL, None)
+
+        await hass.async_create_task(
+            async_load_platform(
+                hass, SWITCH_DOMAIN, DOMAIN, discovery_info, config))
+
+        if include_sensors:
+            response = await get_schedules(device_data.ip_addr, phone_id,
+                                           device_id, device_password)
+
+            if response.found_schedules:
+                discovery_info[DISCOVERY_SCHEDULES] = response.get_schedules
+
+            await hass.async_create_task(
+                async_load_platform(
+                    hass, SENSOR_DOMAIN, DOMAIN, discovery_info,
+                    config))
+
+            async_track_time_interval(hass,
+                                      async_update_schedules_data,
+                                      schedules_scan_interval)
+
+        return None
+
+    async def async_switch_platform_discoverd(platform: str,
+                                              discovery_info: Optional[Dict]
```

Comment:
Name typo:
discovered

Reply:
Thank you.
Fixed in [this](https://github.com/home-assistant/home-assistant/pull/20830/commits/6d478ac3f47207b344a670fc3a04f1811c544a83) commit.